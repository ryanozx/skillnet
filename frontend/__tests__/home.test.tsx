import React from 'react';
import { render } from '@testing-library/react';
import Home from '../pages/index';

jest.mock('next/router', () => require('next-router-mock'));
jest.mock('../withAuthRedirect', () => ({
    preventAuthAccess: jest.fn((Component) => Component),
    requireAuth: jest.fn((Component) => Component),
}));

it('Home page renders correctly', () => {
    const { asFragment } = render(<Home />);
    expect(asFragment()).toMatchSnapshot();
});
