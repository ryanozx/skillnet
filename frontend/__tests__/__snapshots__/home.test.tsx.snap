// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page renders correctly 1`] = `
<DocumentFragment>
  <div
    class="chakra-container css-1qt3bbd"
  >
    <div
      class="chakra-stack css-ilebpv"
    >
      <h2
        class="chakra-heading css-ezczds"
        data-testid="home-page-header"
      >
        SkillNet 
        <br />
        <hr
          aria-orientation="horizontal"
          class="chakra-divider css-nkr49l"
        />
        <span
          class="chakra-text css-1ou2t5u"
          data-testid="home-page-subheader"
        >
          Empower your ideas
        </span>
      </h2>
      <div
        class="chakra-stack css-1pe0sil"
      >
        <a
          class="chakra-link css-0"
          href="/login"
        >
          <button
            class="chakra-button css-xyvrol"
            type="button"
          >
            Log In
          </button>
        </a>
        <a
          class="chakra-link css-0"
          href="/signup"
        >
          <button
            class="chakra-button css-4xx2wk"
            type="button"
          >
            Sign Up
          </button>
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
